tests/validation/tests/v3_api/test_service_discovery.py

```diff
@@ -348,6 +408,106 @@ def test_dns_record_type_selector():
     create_and_validate_dns_record(record, expected_ips)
 
 
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_service_discovery_create(role):
+    """
+    Creates dns record and validates it for different roles passed in parameter
+    @param role: User role in rancher eg. project owner, project member etc
+    """
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    workload = rbac_get_workload()
+    p_client = get_project_client_for_token(project, token)
+
+    record = {"type": "dnsRecord", "targetWorkloadIds": [workload["id"]],
+              "name": random_test_name("record"), "namespaceId": ns.id}
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
```

Comment:
We can simplify this here and in the other cases:

```
with pytest.raises(ApiError) as e:
    dns_record = create_dns_record(record, p_client)

if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
    assert e.value.error.status == 403
    assert e.value.error.code == 'Forbidden'
    return


validate_dns_record_for_workload(workload, 1,
                                 record, p_client=p_client)
p_client.delete(dns_record)
```

Reply:
Raising error without condition can fail for cases where error is not expected.